File: Embassat/Categories/UIView+CADAdditions.m
//
//  UIView+CADAdditions.m
//  YouBarcelona
//
//  Created by Joan Romano on 31/07/13.
//  Copyright (c) 2013 Crows And Dogs. All rights reserved.
//

#import "UIView+CADAdditions.h"

@implementation UIView (CADAdditions)

/**
 Get the current first responder without using a private API
 
 http://stackoverflow.com/questions/1823317/get-the-current-first-responder-without-using-a-private-api
 */
- (UIView *)findFirstResponder
{
    if (self.isFirstResponder)
        return self;
    
    for (UIView *subView in self.subviews)
    {
        UIView *firstResponder = [subView findFirstResponder];
        
        if (firstResponder != nil)
        {
            return firstResponder;
        }
    }
    
    return nil;
}

- (UIScrollView *)findParentScrollView
{
    if ([self isKindOfClass:[UIScrollView class]])
        return (UIScrollView *)self;
    
    if (self.superview)
    {
        return [self.superview findParentScrollView];
    }
    
    return nil;
}

@end

File: Embassat/Categories/UIView+CADAdditions.swift
//
//  UIView+CADAdditions.swift
//  Embassa't
//
//  Created by Joan Romano on 9/24/15.
//  Copyright © 2015 Crows And Dogs. All rights reserved.
//

import UIKit

extension UIView {
    
    func findFirstResponder() -> UIView? {
        if (self.isFirstResponder) {
            return self
        }
        
        for subview in self.subviews {
            if let firstResponder = subview.findFirstResponder() {
                return firstResponder
            }
        }
        
        return nil
    }
    
    func findParentScrollView() -> UIScrollView? {
        if self.isKind(of: UIScrollView.self) {
            return self as? UIScrollView
        }
        
        if let superview = self.superview {
            return superview.findParentScrollView()
        }
        
        return nil
    }
}
